tools/nuget/generate_nuspec_for_native_nuget.py

```diff
@@ -484,7 +484,7 @@ def generate_files(line_list, args):
         files_list.append(
             "<file src="
             + '"'
-            + os.path.join(args.sources_path, "include\\onnxruntime\\core\\providers\\dnnl\\dnnl_provider_factory.h")
+            + os.path.join(args.sources_path, "include\\onnxruntime\\core\\providers\\dnnl\\dnnl_provider_options.h")
```

Comment:
Tensorrt is similar situation as it also have only tensorrt_provider_options.h in the corresponding folder, but it does not expose the file to nuget. It is because there is C API to create provider options. 

For DNNL, the api is like CreateDnnlProviderOptions.

I think you can follow tensorrt in line 481 for the change.